tasks/2017/interactive-bots.md

```diff
@@ -240,6 +240,77 @@ the comment section on GitHub.
 *Completion criteria:* Mentors will check if the new bot fulfills the
 feature requirements and follows the proper directory structure.
 
+### Task Type D: Improve an existing bot
+
+We recommend claiming this task type after completing **Task Type A**
+and **Task Type B**.
+
+Let *bot* and \<bot\> be the bot mentioned in the task that brought you here.
+
+* Locate *bot*. It can be either in [python-zulip-api/zulip_bots/zulip_bots/bots](
+https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots/),
+or in [python-zulip-api/zulip_bots/zulip_bots/bots_unmaintained](
+https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots_unmaintained/).
+If it is in `bots_unmaintained`, move it to `bots`.
+
+* Try to run *bot*. Fix any bugs you notice.
+  - Does *bot* start?
+  - Does *bot* respond as expected?
+  - Does *bot* crash on any input?
+
+* Check the structure of *bot*. Fix any inconsistencies you notice.
+  - Is the directory of *bot* properly structured?
+  - Are the files in *bot*'s directory named correctly? Are they complete? If
+  a bot requires a configuration file, the directory should contain a template `<bot>.conf`.
+  - You can look up the proper directory structure and filenames in our
+  [Writing bots guide](https://www.zulipchat.com/api/writing-bots#adding-a-bot-to-zulip).
+
+* Read through the documentation of *bot*. Fix any inconsistencies you notice.
+  - Does the documentation describe the purpose of *bot*?
+  - Does the documentation give instructions for setting up and using *bot*?
+  - Is the documentation properly structured into the sections "Intro", "Setup",
+  and "Usage"?
+
+* Add and run tests for *bot*. Fix any bugs you notice and aim for 100% test coverage.
+  - If *bot* already has unit tests (these are in a file named `test_<bot>.py`):
+    - In the development environment in your python-zulip-api repo, run
+      ```
+      tools/test-bots <bot> --coverage
+      ```
+      This will run the unit tests for *bot* and verify that it responds correctly to
+      messages. Additionally, it will create a folder `htmlcov`.
+    - Open `htmlcov/index.html` and click on `<bot>.py`. You will see the file, with some
```

Comment:
Wouldn't it make more sense to say click on `zulip_bots/zulip_bots/bots/<bot>/<bot>.py`
![screenshot from 2017-11-17 22 12 12](https://user-images.githubusercontent.com/8913577/32959338-55626cee-cbe7-11e7-97c0-452eed26dd46.png)